docusaurus/docs/tutorials/ui-extensions/register-an-extension.md

```diff
@@ -0,0 +1,233 @@
+
+```tsx
+import { usePluginContext, PluginExtensionPoints } from '@grafana/data';
+
+export const plugin = new AppPlugin().addComponent({
+  title: 'User profile tab',
+  description: '...',
+  targets: [PluginExtensionPoints.UserProfileTab],
+  component: () => {
+    const { meta } = usePluginContext();
+
+    // The `jsonData` property is an object that your plugin can manage
+    // using the Grafana Rest APIs
+    return <div>Plugin specific setting: {meta.jsonData.foo}</div>;
+  },
+});
+```
+
+### Access plugin state in a component
+
+```tsx
+import { PluginExtensionPoints } from '@grafana/data';
+import { MyCustomDataProvider } from './MyCustomDataProvider';
+
+export const plugin = new AppPlugin().addComponent({
+  title: 'User profile tab',
+  description: '...',
+  targets: [PluginExtensionPoints.UserProfileTab],
+  component: () => (
+    <MyCustomDataProvider>
+      <div>Plugin specific setting: {meta.jsonData.foo}</div>
+    </MyCustomDataProvider>
+  ),
+});
+```
+
+### Hide a component in certain conditions
+
+Simply return `null` from your component in order to not render anything and thereby hide the component.
+
+```tsx
+import { usePluginContext, PluginExtensionPoints } from '@grafana/data';
+
+export const plugin = new AppPlugin().addComponent({
+  title: 'User profile tab',
+  description: '...',
+  targets: [PluginExtensionPoints.UserProfileTab],
+  component: () => {
+    const { meta } = usePluginContext();
+
+    // For the sake of the example this condition is relying on a `jsonData` property
+    // that is managed by your plugin
+    if (!meta.jsonData.isExtensionEnabled) {
+      return null;
+    }
+
+    return <div>Plugin specific setting: {meta.jsonData.foo}</div>;
+  },
+});
+```
+
+## Troubleshooting
+
+### My link is not appearing
+
+1. **Check the console logs** - there is a good chance that your link is not appearing due to some validation errors. In this case you should see some relevant logs in your browsers console.
+1. **Check the `targets`** - make sure that you are using the correct extension point ids (always use the `PluginExtensionPoints` enum for Grafana extension points)
+1. **Check the links `configure()` function** - in case your link has a `configure()` function it can happen that it is returning `undefined` under certain conditions, which hides the link.
+1. **Check if you register too many links** - certain extension points limit the number of links allowed per plugin, and in case your plugin registers more than one links for the same extension point there is a chance that some of them are filtered out.
+1. **Check the Grafana version** - link and component extensions are only supported after Grafana version **`>=10.1.0`**, while `addLink()` is only supported in versions **>=`11.1.0`**.
+
```

Comment:
We need examples showing how to support both APIs during a transition phase. Question is if this should be described in a separate page. WDYT @sympatheticmoose @mckn?

Reply:
would it make sense as a migration guide?

Reply:
I think the migration guide is a good place for this.